llvm/lib/Target/SBF/SBFTargetTransformInfo.h

```diff
@@ -71,12 +71,14 @@ class SBFTTIImpl : public BasicTTIImplBase<SBFTTIImpl> {
       TTI::OperandValueInfo Op2Info = {TTI::OK_AnyValue, TTI::OP_None},
     ArrayRef<const Value *> Args = ArrayRef<const Value *>(),
     const Instruction *CxtI = nullptr) {
-      int ISD = TLI->InstructionOpcodeToISD(Opcode);
-      if (ISD == ISD::ADD && CostKind == TTI::TCK_RecipThroughput)
-        return SCEVCheapExpansionBudget.getValue() + 1;
-
-      return BaseT::getArithmeticInstrCost(Opcode, Ty, CostKind, Op1Info,
-                                           Op2Info);
+    int ISDOpcode = TLI->InstructionOpcodeToISD(Opcode);
+    
+    if ((ISDOpcode == ISD::ADD && CostKind == TTI::TCK_RecipThroughput) ||
+        !Ty->isIntOrPtrTy())
+      return SCEVCheapExpansionBudget.getValue() + 1;
```

Comment:
My idea was to use the cheap expansion budget only for floating points, instead of that awkward restriction to additions.

It turns out that is a hack to satisfy the kernel verifier (see the comment in https://github.com/anza-xyz/llvm-project/commit/e52946b9ababcbf8e6f40b1b15900ae2e795a1c6). Although this is not the ideal approach, it save us three CUs in some cases.